### DISCUSSION TITLE
Creating custom object encoder with a primitive array member

### DISCUSSION THREAD

**Discussion by madshov** [uber-go/zap#1099] [Q&A] [answered]
Hi, I'm using `ObjectMarshaler` to create log context for a given struct, so that I be able to use `zap.Object`:
```
logger = logger.With(
        zap.Object("order", order),
)
logger. Info("abc")
```

My `Order` struct contains a `[]string` slice of keywords:
```
type Order struct {
	ID       string
	Price    int64
        Keywords []string
       ...
}
```

My question is how do I include this field in the object encoder for `Order`?
```
func (o Order) MarshalLogObject(enc zapcore.ObjectEncoder) error {
	enc.AddString("id", o.ID)
	enc.AddInt64("price", o.Price)
	enc.AddString("keywords", o.Keywords) // ???
       ...
}

**Comment by abhinav** [comment, marked as answer]
If you have a `[]string` inside a ObjectMarshaler implementation,
you can re-use the [`zap.Strings`](https://pkg.go.dev/go.uber.org/zap#Strings) field constructor
and add it to the encoder with [`Field.AddTo`](https://pkg.go.dev/go.uber.org/zap@v1.21.0/zapcore#Field.AddTo).

```go
func (o Order) MarshalLogObject(enc zapcore.ObjectEncoder) error {
	enc.AddString("id", o.ID)
	enc.AddInt64("price", o.Price)
	zap.Strings("keywords", o.Keywords).AddTo(enc)
       // ...
}
```

Hope this helps!
